llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -1074,10 +1076,12 @@ static MemAccessTy getAccessType(const TargetTransformInfo &TTI,
           II->getArgOperand(1)->getType()->getPointerAddressSpace();
       break;
     default: {
-      MemIntrinsicInfo IntrInfo;
-      if (TTI.getTgtMemIntrinsic(II, IntrInfo) && IntrInfo.PtrVal) {
-        AccessTy.AddrSpace
-          = IntrInfo.PtrVal->getType()->getPointerAddressSpace();
+      std::pair<std::optional<MemIntrinsicInfo>,
+                std::optional<SmallVector<InterestingMemoryOperand, 1>>>
+          MemInfo = TTI.getTgtMemIntrinsic(II);
+      if (MemInfo.first != std::nullopt && MemInfo.first->PtrVal) {
```

Comment:
```suggestion
      if (MemInfo.first.PtrVal) {
```
Check `PtrVal` here too.